src/coreclr/jit/codegenlinear.cpp

```diff
@@ -2059,9 +2063,15 @@ void CodeGen::genSpillLocal(unsigned varNum, var_types type, GenTreeLclVar* lclN
     // but the GTF_SPILL flag records the fact that the register value is going dead.
     if (((lclNode->gtFlags & GTF_VAR_DEF) != 0) || (!varDsc->IsAlwaysAliveInMemory()))
     {
-        // Store local variable to its home location.
-        // Ensure that lclVar stores are typed correctly.
+// Store local variable to its home location.
+// Ensure that lclVar stores are typed correctly.
+#ifdef TARGET_RISCV64
+        assert(!compiler->isSIMDTypeLocalAligned(varNum));
+        GetEmitter()->emitIns_S_R(ins_Store(type, compiler->isSIMDTypeLocalAligned(varNum)), emitTypeSize(type), regNum,
+                                  REG_NA,
+#else
         GetEmitter()->emitIns_S_R(ins_Store(type, compiler->isSIMDTypeLocalAligned(varNum)), emitTypeSize(type), regNum,
+#endif
```

Comment:
Can you add a new RISC-V only function for when the temporary register is necessary, to be used in the places that do not pass `REG_NA`? That way we can avoid adding all these ifdefs that just pass `REG_NA` anyway in shared code.

Reply:
Fixed! Thank you.